### PR TITLE
vyos_facts 2.3

### PR DESCRIPTION
**Pull request by Qalthos**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

vyos_facts

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY
Update vyos_config to LocalAnsibleModule

### CODE REVIEW

File: lib/ansible/modules/network/vyos/vyos_facts.py [hunk 1]
```diff
@@ -15,29 +15,30 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
-ANSIBLE_METADATA = {'status': ['preview'],
-                    'supported_by': 'community',
-                    'version': '1.0'}
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
```
Comment (gundalow) [inline lib/ansible/modules/network/vyos/vyos_facts.py]: Need to check,  though I believe we support `*_facts` modules.

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/vyos.py [hunk 2]
```diff
@@ -26,87 +26,63 @@
...
-    TERMINAL_LENGTH = os.getenv('ANSIBLE_VYOS_TERMINAL_LENGTH', 10000)
+    for cmd in commands:
+        rc, out, err = module.exec_command(cmd)
+        if check_rc and rc != 0:
+            module.fail_json(msg=err, rc=rc)
+        responses.append(out)
+    return responses
 
+def load_config(module, commands, commit=False, comment=None, save=False):
+    assert isinstance(commands, list), 'commands must be a list'
```
Comment (privateip) [inline lib/ansible/module_utils/vyos.py]: remove assert and use `to_list` from `ansible.module_utils.network_common`

File: lib/ansible/module_utils/vyos.py [hunk 3]
```diff
@@ -26,87 +26,63 @@
...
+    _DEVICE_CONFIG = out.strip()
+    return _DEVICE_CONFIG
 
-    CLI_ERRORS_RE = [
-        re.compile(r"\n\s*Invalid command:"),
-        re.compile(r"\nCommit failed"),
-        re.compile(r"\n\s+Set failed"),
-    ]
+def run_commands(module, commands, check_rc=True):
+    assert isinstance(commands, list), 'commands must be a list'
```
Comment (privateip) [inline lib/ansible/module_utils/vyos.py]: remove assert and use `to_list` from `ansible.module_utils.network_common`

File: lib/ansible/module_utils/vyos.py [hunk 4]
```diff
@@ -26,87 +26,63 @@
...
 #
 
-import re
-import os
+_DEVICE_CONFIG = None
 
-from ansible.module_utils.network import NetworkModule, NetworkError
-from ansible.module_utils.network import register_transport, to_list
-from ansible.module_utils.shell import CliBase
+def get_config(module):
```
Comment (privateip) [inline lib/ansible/module_utils/vyos.py]: port `get_config()` from vyos2 to vyos

File: lib/ansible/modules/network/vyos/vyos_facts.py (hunk 1 above)
Comment (privateip) [reply lib/ansible/modules/network/vyos/vyos_facts.py]: core supported

**Review by privateip — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/vyos/vyos_facts.py (hunk 1 above)
Comment (Qalthos) [reply lib/ansible/modules/network/vyos/vyos_facts.py]: I was wondering about that.

File: lib/ansible/module_utils/vyos.py (hunk 4 above)
Comment (Qalthos) [reply lib/ansible/module_utils/vyos.py]: I don't know how that got lost.

**Pull request merged**